extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -102,36 +108,44 @@ class LifecycleTask extends BackbeatTask {
                 return next(null, data);
             }),
             (data, next) => {
+                // sorted newest to oldest
+                // NOTE: parentDate can be undefined
+                let parentDate = prevDate;
```

Comment:
Rename to `staleDate` or `noncurrentDate`? Seems like what it represents when used to set a field in `sortedVersions` is the date of the nearest newer version, where a parent would be an ancestor (older) version, just a semantic issue but may be clearer using another naming than parent.

Reply:
Ah good call, I agree, use of parent does sound like ancestor